uxarray/core/utils.py

```diff
@@ -3,6 +3,41 @@
 from uxarray.io.utils import _get_source_dims_dict, _parse_grid_type
 
 
+def _open_dataset_with_fallback(filename_or_obj, chunks=None, **kwargs):
+    """Internal utility function to open datasets with fallback to netcdf4 engine.
+
+    Attempts to use Xarray's default read engine first, which may be "h5netcdf"
+    or "scipy" after v2025.09.0. If that fails (typically for h5-incompatible files),
+    falls back to using the "netcdf4" engine.
+
+    Parameters
+    ----------
+    filename_or_obj : str, Path, file-like or DataStore
+        Strings and Path objects are interpreted as a path to a netCDF file
+        or an OpenDAP URL and opened with python-netCDF4, unless the filename
+        ends with .gz, in which case the file is gunzipped and opened with
+        scipy.io.netcdf (only netCDF3 supported).
+    chunks : int, dict, 'auto' or None, optional
+        If chunks is provided, it is used to load the new dataset into dask
+        arrays.
+    **kwargs
+        Additional keyword arguments passed to xr.open_dataset
+
+    Returns
+    -------
+    xr.Dataset
+        The opened dataset
+    """
+    try:
+        # Try opening with xarray's default read engine
+        return xr.open_dataset(filename_or_obj, chunks=chunks, **kwargs)
+    except Exception:
+        # If it fails, use the "netcdf4" engine as backup
+        return xr.open_dataset(
+            filename_or_obj, engine="netcdf4", chunks=chunks, **kwargs
```

Comment:
Add a safeguard for the case where `engine` may be provided within the `kwargs` to prevent duplicate parameters.

Reply:
Great catch!